src/views/wiki/edition/OutingEditionView.vue

```diff
@@ -190,18 +237,168 @@ export default {
+          // and if there is no specific bbox expected
+          this.bbox = bbox;
+          // then do not load possible routes
+          return;
+        }
+      }
+
+      if (bbox.join(',') !== this.bbox?.join(',')) {
+        this.bbox = bbox;
+        this.updateRoutes('bbox', false);
+      }
+    },
+
+    // printBbox(reason, bbox) {
+    //   bbox = bbox ? bbox : this.bbox;
+    //   console.log([reason, (bbox[2] - bbox[0])/1000, (bbox[3] - bbox[1])/1000]);
+    // },
+
+    fitMapToDocuments() {
+      this.bbox = this.$refs.mapInput.fitMapToDocuments();
+    },
+
+    updateRoutes(reason, fitMap) {
+      if (!this.bbox) {
+        return reason;
+      }
+
+      const query = { limit: 50, bbox: this.bbox.join(',') };
+
+      if (this.routeTitle) {
+        query.q = this.routeTitle;
+      }
+
+      if (this.document.activities.length !== 0) {
+        query.act = this.document.activities.join(',');
+      }
+
+      const promise = c2c.route.getAll(query);
+
+      promise.then((response) => this.computePossibleRoutes(response.data.documents, response.data.total, fitMap));
+
+      return promise;
+    },
+
+    clearPossibleRoutes() {
+      this.computePossibleRoutes([], 0, false);
+    },
+
+    computePossibleRoutes(routes, total, fitMap) {
+      this.showMoreResultsBanner = total !== routes.length;
+
+      // always add actual associations to possible routes
+      for (const route of this.document.associations.routes) {
+        if (routes.filter((r) => r.document_id === route.document_id).length === 0) {
+          routes.push(route);
+        }
+      }
+
+      // sort by title
+      routes.sort((a, b) =>
+        this.$documentUtils.getDocumentTitle(a).localeCompare(this.$documentUtils.getDocumentTitle(b))
+      );
+
+      // assign
+      this.possibleRoutes = routes;
+
+      // recenter the map to possible routes
+      if (fitMap) {
+        this.$nextTick(this.fitMapToDocuments);
+      }
+    },
+
+    changeRouteAssociation(addRoute, route) {
+      if (!addRoute && !this.routeIsAssociated(route.document_id)) {
+        return;
+      } else if (addRoute && this.routeIsAssociated(route.document_id)) {
+        return;
+      }
+
+      const routes = this.document.associations.routes;
+
+      if (addRoute) {
+        routes.push(route);
+
+        if (routes.length === 1) {
+          this.possibleRoutes = [];
+
+          this.$nextTick(() => {
+            // for first association, force the localization to be the route localization
+            this.$documentUtils.propagateProperties(this.document, route);
+            this.fitMapToDocuments();
+            this.updateRoutes('First route', false);
+          });
+        } else {
+          this.$documentUtils.propagateProperties(this.document, route);
+        }
+      } else {
+        this.document.associations.routes = routes.filter((r) => r.document_id !== route.document_id);
+
+        if (this.document.associations.routes.length === 0) {
```

Comment:
```suggestion
        if (!this.document.associations.routes.length) {
```

Reply:
Done. 

Though, could you explain why the second line is better? I really feel that the first one is much more clear, and if there is a performance issue, it should be negligible for a line executed only once, no?

Actually, if it's a good practice to use boolean operator instead of `===0`, we should consider implementing it in the linter, it'll save some time and energy for PR.